src/main/scala/org/apache/daffodil/DaffodilPlugin.scala

```diff
@@ -355,8 +355,11 @@ object DaffodilPlugin extends AutoPlugin {
           "<=3.10.0" -> "2.12"
         )
         val crossVersion = filterVersions(binDaffodilVersion, daffodilToCrossVersion).head
-        val dafDep = ("org.apache.daffodil" % "daffodil-japi" % binDaffodilVersion % cfg)
-          .withCrossVersion(CrossVersion.constant(crossVersion))
+        val dafDep = {
+          val dep = if (crossVersion == "3") ("org.apache.daffodil" % "daffodil-core" % binDaffodilVersion % cfg)
+                    else ("org.apache.daffodil" % "daffodil-japi" % binDaffodilVersion % cfg)
+          dep.withCrossVersion(CrossVersion.constant(crossVersion))
+        }
```

Comment:
Maybe we should use the filterVersion thing for these deps. So replace daffodilToCrossVersion with something like this:

```scala
val daffodilToPackageBinDep = Map(
  ">=4.0.0 " -> "org.apache.daffodil" % "daffodil-core" % (binDaffodilVersion + "_3") % cfg,
  "=3.11.0 " -> "org.apache.daffodil" % "daffodil-japi" % (binDaffodilVersion + "_2.13") % cfg,
  "<=3.10.0" -> "org.apache.daffodil" % "daffodil-japi" % (binDaffodilVersion + "_2.12") % cfg
)
val dafDep = filterVersion(binDaffodilVersion, daffodilToPackageBinDep)
...
```

That feels a bit cleaner to me, and is also more easily extendable in the future if we need to add additional version specific dependencies. It also feels a bit easier to see exactly what dependency different versions of daffodil use.